hedera-node/src/main/java/com/hedera/services/state/logic/StandardProcessLogic.java

```diff
@@ -86,42 +86,64 @@ public StandardProcessLogic(
 	public void incorporateConsensusTxn(SwirldTransaction platformTxn, Instant consensusTime, long submittingMember) {
 		try {
 			final var accessor = expandHandleSpan.accessorFor(platformTxn);
-			Instant effectiveConsensusTime = consensusTime;
-			if (accessor.canTriggerTxn()) {
-				final var offset = dynamicProperties.triggerTxnWindBackNanos();
-				effectiveConsensusTime = consensusTime.minusNanos(offset);
-			}
 
-			if (!invariantChecks.holdFor(accessor, effectiveConsensusTime, submittingMember)) {
+			if (!invariantChecks.holdFor(accessor, consensusTime, submittingMember)) {
 				return;
 			}
 
-			sigImpactHistorian.setChangeTime(effectiveConsensusTime);
-			expiries.purge(effectiveConsensusTime.getEpochSecond());
+			consensusTimeTracker.reset(consensusTime);
+
+			sigImpactHistorian.setChangeTime(consensusTime);
+			expiries.purge(consensusTime.getEpochSecond());
 			sigImpactHistorian.purge();
 			recordStreaming.resetBlockNo();
 
-			doProcess(submittingMember, consensusTime, effectiveConsensusTime, accessor);
+			doProcess(submittingMember, consensusTimeTracker.isFirstUsed()
+					? consensusTimeTracker.nextTransactionTime(true)
+					: consensusTimeTracker.firstTransactionTime(), accessor);
+
+			processScheduledTransactions(consensusTime, submittingMember);
 
-			autoRenewal.execute(recordsHistorian.nextFollowingChildConsensusTime());
+			autoRenewal.execute(consensusTime);
 		} catch (InvalidProtocolBufferException e) {
 			log.warn("Consensus platform txn was not gRPC!", e);
 		} catch (Exception internal) {
 			log.error("Unhandled internal process failure", internal);
 		}
 	}
 
+	private void processScheduledTransactions(Instant consensusTime, long submittingMember) {
+		TxnAccessor triggeredAccessor = null;
+		while (true) {
```

Comment:
Would feel better with a,
```
do {
    ...
} while (hasMore); 
```
construct 😉

Reply:
Done, and limit to loops added.

Reply:
Awesome!